src/vs/workbench/browser/parts/editor/editorStatus.ts

```diff
@@ -671,7 +671,7 @@ export class EditorStatus extends Disposable implements IWorkbenchContribution {
 			}));
 
 			// Hook Listener for Selection changes
-			this.activeEditorListeners.add(activeCodeEditor.onDidChangeCursorPosition(() => {
+			this.activeEditorListeners.add(Event.defer(activeCodeEditor.onDidChangeCursorPosition, this._store)(() => {
```

Comment:
We might want to call here `Event.defer(..., this.activeEditorListeners)` instead of `Event.defer(..., this._store_)` but I have to say I managed once a few months ago to understand how these event utilities cause leakages, but in the meantime I forgot again. Let's ask @jrieken or @joaomoreno for advice, is a `DisposableStore` needed here or not?

Reply:
Yeah, the warning aligns with your intuition:

> *NOTE* that this function returns an `Event` and it MUST be called with a `DisposableStore` whenever the returned event is accessible to "third parties", e.g the event is a public property.

Reply:
So @Tyriar the request here is to use `Event.defer(activeCodeEditor.onDidChangeCursorPosition)`